ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschema.py

```diff
@@ -204,11 +204,27 @@ def subschema(self, datapath: list, schema: dict = None):
         if not isinstance(key, str):
             raise AvdSchemaError(f"All datapath items must be strings. Got {type(key)}")
 
-        if schema["type"] == "dict" and key in schema.get("keys", []):
-            return self.subschema(datapath[1:], schema["keys"][key])
+        if schema["type"] == "dict":
+            if key in schema.get("keys", []):
+                resolved_schema = self.simple_resolve(schema["keys"][key])
+                return self.subschema(datapath[1:], resolved_schema)
+            if key in schema.get("dynamic_keys", []):
+                resolved_schema = self.simple_resolve(schema["dynamic_keys"][key])
+                return self.subschema(datapath[1:], resolved_schema)
 
         if schema["type"] == "list" and key in schema.get("items", {}).get("keys", []):
             return self.subschema(datapath[1:], schema["items"]["keys"][key])
 
         # Falling through here in case the schema is not covering the requested datapath
         raise AvdSchemaError(f"The datapath '{datapath}' could not be found in the schema")
+
+    def simple_resolve(self, schema):
+        # Get fully resolved schema (where all $ref has been expanded recursively)
+        # Performs inplace update of the argument so we give an empty dict.
+        # By default it will resolve the full schema
```

Comment:
```suggestion
    def simple_resolve(self, schema: dict):
        """
        Get fully resolved schema (where all $ref has been expanded recursively)
        Performs inplace update of the argument so we give an empty dict.
        By default it will resolve the full schema
        """
```

Maybe change the name to `get_resolved_schema`

Reply:
done